src/InteractionModWorkNborhood.H

```diff
@@ -0,0 +1,102 @@
+        AMREX_ALWAYS_ASSERT(random_travel_ptr[infectious_i] < 0 && random_travel_ptr[susceptible_i] < 0);
+
+        //infect *= i_mask;
+        //infect *= j_mask;
+        ParticleReal prob = 1.0_prt;
+        // define neighbordhood based on whether agent is in home or work community
+        int nborhood_infectious_i = nborhood_ptr[infectious_i];
+        int nborhood_susceptible_i = nborhood_ptr[susceptible_i];
+        if (age_group_ptr[infectious_i] > 1) nborhood_infectious_i = work_nborhood_ptr[infectious_i];
+        if (age_group_ptr[susceptible_i] > 1) nborhood_susceptible_i = work_nborhood_ptr[susceptible_i];
+
+        // school < 0 means a child normally attends school, but not today
+        // Should always be in the same community = same cell
+        // Neighborhood?
+        if (nborhood_infectious_i == nborhood_susceptible_i) {
+            if (school_ptr[infectious_i] < 0) { // not attending school, use _SC contacts
+                prob *= 1.0_prt - infect * a_lparm->xmit_hood_SC[age_group_ptr[susceptible_i]] * a_social_scale;
+            } else {
+                prob *= 1.0_prt - infect * a_lparm->xmit_hood[age_group_ptr[susceptible_i]] * a_social_scale;
+            }
+        } else { // Community?
+            if (school_ptr[infectious_i] < 0) { // not attending school, use _SC contacts
+                prob *= 1.0_prt - infect * a_lparm->xmit_comm_SC[age_group_ptr[susceptible_i]] * a_social_scale;
+            } else {
+                prob *= 1.0_prt - infect * a_lparm->xmit_comm[age_group_ptr[susceptible_i]] * a_social_scale;
+            }
+        }
+        return prob;
+    }
+};
+
+template <typename PTDType>
+struct WorkNborhoodCandidate {
+    AMREX_GPU_HOST_DEVICE
+    bool operator() (const int idx, const PTDType& ptd) const noexcept {
+        return !isHospitalized(idx, ptd) && !ptd.m_idata[IntIdx::withdrawn][idx] && ptd.m_idata[IntIdx::random_travel][idx] < 0;
+    }
+};
+
+
+/*! \brief Class describing agent interactions in the neighborhood/community */
+template <typename PCType, typename PTDType, typename PType>
+class InteractionModWorkNborhood : public InteractionModel<PCType, PTDType, PType>
+{
+    public:
+
+        /*! \brief null constructor */
+        InteractionModWorkNborhood (bool _fast_bin) : InteractionModel<PCType, PTDType, PType>(_fast_bin) {}
+
+        /*! \brief default destructor */
+        virtual ~InteractionModWorkNborhood() = default;
+
+        /*! \brief Simulate agent interaction in the neighborhood/community */
+        virtual void interactAgents (PCType& agents, MultiFab&) override {
+            // passing -1 for the binning group indicates bin only by community (cell)
+            interactAgentsImpl<InteractionModWorkNborhood<PCType, PTDType, PType>, PCType, PTDType,
+                                WorkNborhoodCandidate<PTDType>,
+                                BinaryInteractionWorkNborhood<PTDType>>(*this, agents, -1);
```

Comment:
Why `-1` here? When I make separate timers for each type of interaction, it looks like much more time is spent in this one than all the others. Can't we bin this by `IntIdx::work_nborhood`?

Reply:
I see... it's because we also account for general community spread in this routine.

Reply:
Yes, this is the slowest interaction, since it's the full community.